src/peer/identity.js

```diff
@@ -28,25 +27,24 @@ function loadIdentity (filePath: string): Promise<PeerId> {
 }
 
 
-function loadOrGenerateIdentity(filePath: string): PeerId {
-  let peerId
-  try {
-    loadIdentity(filePath)
-  } catch(err) {
-    console.log(`Could not load from ${filePath}, generating new PeerId...`)
-    peerId = generateIdentity()
-    saveIdentity(peerId, filePath)
-  }
-
-  return peerId
+function loadOrGenerateIdentity(filePath: string): Promise<PeerId> {
+    return loadIdentity(filePath)
+      .catch(err => {
+        console.log(`Could not load from ${filePath}, generating new PeerId...`)
+        return generateIdentity()
```

Comment:
I would really like to add a type check on this error so we don't happily try to create a new key every run if e.g. we don't have permissions to `~/.mediachain`